platforms/core-execution/build-cache-http/build.gradle.kts

```diff
@@ -17,7 +18,6 @@ dependencies {
     api(projects.coreApi)
     api(projects.resourcesHttp)
 
-    implementation(projects.internalInstrumentationApi)
     implementation(projects.core)
     implementation(projects.logging)
     implementation(projects.resources)
```

Comment:
I think we should straighten this out. There is no reason for instrumentation to depend on `:base-services`, and indeed we should break that project up and stop depending on it exactly because it being such a "god-project" causes other improvements to become impossible.

IIUC, only `:base-asm` depends on `:base-services`. There are three things it depends on:

- `AsmConstants.ASM_LEVEL` -- we should just inline this, there is no reason why we'd need to share this setting the way we do now.
- `ClassLoaderUtils.defineDecorator()` is only used in `:base-asm` and should be moved there.
- `JavaReflectionUtil.getWrapperTypeForPrimitiveType()` -- I'd be okay with inlining this, too; it's not like this code is expected to change anytime soon.

Some more options:

- introduce a new `:java-reflection` module, and move all of `org.gradle.internal.reflect` to it, and make anyone wanting to do reflection depend on it, including `:base-asm`
- move all of `org.gradle.internal.reflect` to `:base-asm`; it's not a great solution, but I think it's already better than having anyone needing reflection be forced to depend on all of `:base-services`
- move all of `org.gradle.internal.reflect` to `:base-asm` and rename the project to `:java-reflection`.

Doing either of the above would allow us to stop depending on `:base-services`, and implement the right solution to the above problem.